exchanges/stream/websocket.go

```diff
@@ -457,10 +478,7 @@ func (w *Websocket) FlushChannels() error {
 	}
 
 	err := w.Shutdown()
```

Comment:
By moving `FlushChannels` into `Connect`, the function is under a lock when it wasn't before. So if the two `if` statements above aren't met, the `Shutdown` function will be called which tries to lock, this will cause a deadlocked websocket, which does occur for Gemini. While its true that Gemini needs to be updated 😄 this scenario is still present and will cause the deadlock